hub/subscribe.go

```diff
@@ -87,16 +87,21 @@ func (h *Hub) initSubscription(w http.ResponseWriter, r *http.Request) (*Subscri
 	authorizedAlltargets, authorizedTargets := authorizedTargets(claims, false)
 	subscriber := NewSubscriber(authorizedAlltargets, authorizedTargets, topics, rawTopics, templateTopics, retrieveLastEventID(r))
 
-	if subscriber.LastEventID != "" {
-		h.sendMissedEvents(w, r, subscriber)
-	}
-
 	// Create a new channel, over which the hub can send can send updates to this subscriber.
 	updateChan := make(chan *serializedUpdate)
 
 	// Add this client to the map of those that should
 	// receive updates
+
+	useHistory := subscriber.LastEventID != ""
+	if useHistory {
+		h.history.Lock()
```

Comment:
this locks all writes until this subscriber reads the last missed event. This can be a performance issue if the history is huge or network with that subscriber is slow.

Reply:
and that also prevents 2 concurrent reads